File: typescript/norepeatlongeststring.ts
function getmaxchildstringlength(str: String): number {
    let left: number = 0;
    let res: number = 0;
    let dic: { [id: string]: number} = {};
    for(let right = 0; right < str.length; right++) {
        if(typeof dic[str[right]] !== 'undefined') {
            left = dic[str[right]];
        }
        dic[str[right]] = right + 1;
        res = res > (right - left + 1) ? res : (right - left + 1)
    }
    return res
}
console.log(getmaxchildstringlength('abcdea1234'))

File: python/norepeatlongeststring.py
def getmaxchildstringlength(string):
    left, res = 0, 0
    dic = {}
    for right in range(len(string)):
        if dic.has_key(string[right]):
            left = dic[string[right]]  # max(dic[string[right]], left)
        dic[string[right]] = right + 1
        res = max(right - left + 1, res)
    return res


print(getmaxchildstringlength('abcdea1234'))
